### PR TITLE
[CHG] b_shift: planning generation more generic

### PR DESCRIPTION
**Pull request by MatteoLdx**
This change does not change the algorithm of planning generation. But it changes what functions returns to let them be used elsewhere and be overwritten in other modules.

Credits to @elioexe

### CODE REVIEW

File: beesdoo_shift/models/planning.py [hunk 1]
```diff
@@ -112,7 +112,8 @@ def _generate_next_planning(self):
             )
             return
 
-        planning.task_template_ids._generate_task_day()
+        task_recset = planning.task_template_ids._generate_task_day()
+        task_recset = task_recset.create_all()
```
Comment (carmenbianca) [inline beesdoo_shift/models/planning.py]: I'm not entirely sure this is a good pattern. Doing one thing should ideally be done in one line.

Would it be possible to do something like this:

- `_generate_task_day()` exhibits the behaviour of the two lines created above.
- A new method is created that exhibits the behaviour of `_generate_task_day()` sans `create_all()`. This method is called internally by `_generate_task_day()`.

Before implementing that suggestion, verify that it's actually a good idea :) I don't know the context behind this change.

**Comment by victor-champonnois** [comment]
@carmenbianca In order to fix the test error that was due to the EMC form, I merged [this PR](https://github.com/beescoop/Obeesdoo/pull/263). However the test still fails, it behaves as with the old 12.0 code. Does this branch needs to be rebased on top of 12.0 for it to work ?

**Comment by carmenbianca** [comment]
I don't think it should. The CI automagically merges the main branch before testing. Not sure why this is failing. Maybe a cache miss? It wouldn't hurt to try rebasing on 12.0.

File: beesdoo_shift/models/task.py [hunk 2]
```diff
@@ -442,3 +442,20 @@ def _get_counter_date_state_change(self, new_state):
...
+                "task_type_id": rec.task_type_id.id,
+                "super_coop_id": rec.super_coop_id.id,
+                "worker_id": rec.worker_id.id,
+                "is_regular": rec.is_regular,
+                "start_time": rec.start_time,
+                "end_time": rec.end_time,
+                "state": "open",
+            }
+            tasks |= rec.create(data)
+        return tasks
```
Comment (remytms) [inline beesdoo_shift/models/task.py]: You can drop this.

**Comment by remytms** [comment]
@carmenbianca Mattéo takes your comment into account. Does this solution sound ok to you ? :)

**Review by remytms — APPROVED** [review]

**Review by carmenbianca — APPROVED** [review]
LGTM.

As a minor comment, `get_task_day` is not used. Do we anticipate use for it?

**Comment by remytms** [comment]
`get_task_day` will be used in the future, by other modules : beesdoo_shift_swap and beesdoo_website_shift. And it can also be used in this module to compute future next shift of a worker without writing it in the database. :)

**Comment by remytms** [comment]
@MatteoLdx You can squash. Then we will put that in test. :)

**Comment by remytms** [comment]
@MatteoLdx How it has already been squash. Nice. :)

**Review by remytms — APPROVED** [review]

**Comment by remytms** [comment]
@MatteoLdx Can you try to rebase it on top of 12.0 because it seams to have some conflict with the 12.0 branch.

**Comment by MatteoLdx** [comment]
> @MatteoLdx Can you try to rebase it on top of 12.0 because it seams to have some conflict with the 12.0 branch.

I resolved the conflict directly on GitHub and it merged 12.0 into this branch, is that okay ?

**Review by remytms — APPROVED** [review]

**Comment by remytms** [comment]
@MatteoLdx We will not merge it now, until this has been tested on the test server. :)

**Comment by carmenbianca** [comment]
> I resolved the conflict directly on GitHub and it merged 12.0 into this branch, is that okay ?

I wouldn't do that. It makes future migration more difficult. Rebasing and force-pushing is easier and cleaner in the history.

**Comment by MatteoLdx** [comment]
> > I resolved the conflict directly on GitHub and it merged 12.0 into this branch, is that okay ?
> 
> I wouldn't do that. It makes future migration more difficult. Rebasing and force-pushing is easier and cleaner in the history.

I agree, in fact I didn't know that GitHub would do that. Should I reset and rebase instead ?

**Comment by carmenbianca** [comment]
Yes, rebase on top of 12.0 and drop [fbe0122](https://github.com/beescoop/Obeesdoo/pull/311/commits/fbe012296c927265caae9b00ea1adea3fec8b3d0)

**Comment by MatteoLdx** [comment]
> Yes, rebase on top of 12.0 and drop [fbe0122](https://github.com/beescoop/Obeesdoo/pull/311/commits/fbe012296c927265caae9b00ea1adea3fec8b3d0)

Done !

**Comment by MatteoLdx** [comment]
Merged with #309

**Pull request closed without merging**